### DISCUSSION TITLE
Access denied for user 'firefly'@'localhost'

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [firefly-iii/firefly-iii#8667] [Firefly III] [answered]
Hello,

This seems to be a common issue but none of the fixes have helped me. I'm attempting to do a fresh install of Firefly III using Docker. I have downloaded all files, renamed the .env and .db.env files, and provided the cron token, db password, and app encryption key.

None of the three values have the # character, and I've even gone to the point of making sure the db password is 20 characters long and does not contain *any* special characters.

With the exception of changing my timezone to `America/Chicago`, all three files are stock.

Sanity checks:

- Db Password is the same in .env and .db.env as ensured by copy-pasting. There are no additional space characters or the like, and no special characters are present.
- Any changes to these files was followed by deleting Firefly III from Docker Desktop and then re-running `docker compose -f docker-compose.yml up -d --pull=always` to ensure there was nothing cached anywhere.
- Executing `docker exec -it firefly_iii_db mariadb --user firefly -pPassword` returns the same "Access denied for user" error: `ERROR 1045 (28000): Access denied for user 'firefly'@'localhost' (using password: YES)`
- With the exception of providing the cron token, there are no changes to the docker-compose.yml file.

Any ideas what is going on here?

**Comment by JC5** [comment, marked as answer]
I suspect the volumes still exist. Delete those too.

**Reply by TheBrambleShark** [reply]
Seems deleting from Docker Desktop does not delete the volumes. Running `docker volume ls` revealed exactly what you suspected. I used `docker volume rm` to remove the volumes, then ran `docker compose`, and Firefly is up and running.

Thanks for the help!

**Reply by JC5** [reply]
Indeed. That's a good thing though, if you ever upgrade you'll be happy that your volumes are still there 😉 .

Having said that, if you turn off and remove everything, I believe you can also "prune" unused volumes.
